pages/dnd/update-layout-test.page.tsx

```diff
@@ -0,0 +1,73 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Header } from "@cloudscape-design/components";
+import { useState } from "react";
+import { Board, BoardItem, BoardProps, ItemsPalette, ItemsPaletteProps } from "../../lib/components";
+import PageLayout from "../app/page-layout";
+import { ScreenshotArea } from "../screenshot-area";
+import { boardI18nStrings, boardItemI18nStrings, itemsPaletteI18nStrings } from "../shared/i18n";
+import { ItemData } from "../shared/interfaces";
+import classnames from "./engine.module.css";
+
+export default function () {
+  const [boardItems, setBoardItems] = useState<readonly BoardProps.Item<ItemData>[]>([]);
+  const [paletteItems, setPaletteItems] = useState<readonly ItemsPaletteProps.Item<ItemData>[]>([
+    {
+      id: "demo",
+      definition: { defaultColumnSpan: 4, minRowSpan: 8 },
+      data: { title: "", description: "", content: "" },
+    },
+  ]);
+
+  const [boardWidth, setBoardWidth] = useState(800);
+  let timeoutId: null | number = null;
+
+  return (
+    <ScreenshotArea>
+      <PageLayout header={<Header variant="h1">In-transition layout update test page</Header>}>
+        <div className={classnames[`layout-grid`]}>
+          <div style={{ width: boardWidth }}>
+            <Board
+              i18nStrings={boardI18nStrings}
+              items={boardItems}
+              renderItem={(item) => (
+                <BoardItem header={<Header>{item.data.title}</Header>} i18nStrings={boardItemI18nStrings}>
+                  {item.data.content}
+                </BoardItem>
+              )}
+              onItemsChange={({ detail: { items, addedItem } }) => {
+                setBoardItems(items);
+                if (addedItem) {
+                  setPaletteItems((paletteItems) => paletteItems.filter((item) => item.id !== addedItem.id));
+                }
+              }}
+              empty="No items"
+            />
+          </div>
+
+          <div className={classnames.palette}>
+            <Header>Add widgets</Header>
+            <ItemsPalette
+              items={paletteItems}
+              renderItem={(item, context) => {
+                if (context.showPreview === true && !timeoutId) {
+                  timeoutId = setTimeout(() => {
+                    setBoardWidth(1000);
+                  }, 0);
```

Comment:
Having side effects inside render functions is a bad practice (even if you are trying to reproduce a bug). I think, you could reproduce the same issue with something like `<div className={classnames['layout-grid']} onMouseDown={() => /*setTimeout here*/}>`

Reply:
That was my first idea as well but the mouse-down event it intercepted by the drag handle so I can't capture it from the outside.

Reply:
What does it mean intercepted?